app/scripts/controllers/swaps.js

```diff
@@ -187,15 +183,21 @@ export default class SwapsController {
         this.setSwapsErrorKey(QUOTES_NOT_AVAILABLE_ERROR)
       } else {
         const {
-          topAggId: topAggIdAfterGasEstimates,
-          isBest: isBestAfterGasEstimates,
+          topAggId: topAggIdAfterModifications,
+          isBest: isBestAfterModifications,
         } = await this._findTopQuoteAggId(newQuotes)
-        topAggId = topAggIdAfterGasEstimates
-        if (isBestAfterGasEstimates) {
-          newQuotes = mapValues(newQuotes, (quote) => ({ ...quote, isBestQuote: false }))
-          newQuotes[topAggId].isBestQuote = true
-        }
+        topAggId = topAggIdAfterModifications
+        isBest = isBestAfterModifications
       }
+    } else {
+      topAggIdData = await this._findTopQuoteAggId(newQuotes)
+      topAggId = topAggIdData.topAggId
+      isBest = topAggIdData.isBest
+    }
+
+    if (isBest) {
+      newQuotes = mapValues(newQuotes, (quote) => ({ ...quote, isBestQuote: false }))
```

Comment:
This step is no longer required. It was only needed because we were finding the best top quote _twice_, so we had to _unset_ the first one after we got the second one. But now we're only finding it once.
```suggestion
```